### PR TITLE
feat: Upgrade AWS ESB CSI and switch to using Helm chart

### PR DESCRIPTION
**Pull request by jimmidyson**
This will make it easier to adopt the HelmAddon strategy for deploying
the AWS ESB CSI addon.

Depends on #392

### CODE REVIEW

File: charts/capi-runtime-extensions/templates/csi/aws-ebs/manifests/aws-ebs-csi-configmap.yaml [hunk 1]
```diff
@@ -1045,7 +1045,12 @@ data:
     kind: ServiceAccount
     metadata:
       labels:
+        app.kubernetes.io/component: csi-driver
+        app.kubernetes.io/instance: aws-ebs-csi-driver
+        app.kubernetes.io/managed-by: Helm
```
Comment (dkoshkin) [inline charts/capi-runtime-extensions/templates/csi/aws-ebs/manifests/aws-ebs-csi-configmap.yaml:1050]: I noticed these Helm labels in other addons, I guess its not a problem for these to be here if not actually managed by helm? I also wonder if there is a helm controller if it tries to "adopt" them.

Something to test later.

**Review by dkoshkin — APPROVED** [review]
Thanks!

File: charts/capi-runtime-extensions/templates/csi/aws-ebs/manifests/aws-ebs-csi-configmap.yaml (hunk 1 above)
Comment (jimmidyson) [reply charts/capi-runtime-extensions/templates/csi/aws-ebs/manifests/aws-ebs-csi-configmap.yaml:1050]: Yes that should make it possible. But yes good to test and see what happens.

Comment (jimmidyson) [reply charts/capi-runtime-extensions/templates/csi/aws-ebs/manifests/aws-ebs-csi-configmap.yaml:1050]: This is the kind of thing we should be able to do via e2e tests pretty easily I hope 🤞

Comment (jimmidyson) [reply charts/capi-runtime-extensions/templates/csi/aws-ebs/manifests/aws-ebs-csi-configmap.yaml:1050]: Opened #394

**Pull request merged**